### PR TITLE
Fix broken NMM build caused by FARMS mods

### PR DESCRIPTION
**Pull request by davegill**
TYPE: bug fix

KEYWORDS: FARMS, NMM

SOURCE: internal

DESCRIPTION OF CHANGES:
Problem:
The NMM HWRF code would not build after the FARMS commit. A few variables passed 
into the radiation driver are ARW specific, so the call from the NMM core did not have:
MUT or DNW (both input associated with ARW only), or the six additional computed 
radiation output fields (sw*2).

Solution:
1. Make the ARW-specific input fields and the additional radiation output fields OPTIONAL.
2. Use an IF(PRESENT(xxx)) before using any of the fields.

These problems were introduced with commit 3c71dd1de6e0df0, PR #1025 "Adding a fast 
surface shortwave parameterization (FARMS)".

LIST OF MODIFIED FILES:
modified:   phys/module_radiation_driver.F
modified:   dyn_em/module_first_rk_step_part1.F

TESTS CONDUCTED:
1. Without mods, NMM HWRF cannot build.
2. With mods, NMM HWRF builds and runs successfully.
3. With mods, ARW still works.

### CODE REVIEW

**Comment by davegill** [comment]
@weiwangncar @dudhia @llpcarson 
Folks,
I just realized that I can probably remove the ifdef mods inside of first_rk_step_part1. I declare the variables as OPTIONAL, and they would always be present with ARW builds.

File: dyn_em/module_first_rk_step_part1.F [hunk 1]
```diff
@@ -308,10 +308,12 @@ SUBROUTINE first_rk_step_part1 (   grid , config_flags              &
      &        ,CAM_ABS_FREQ_S=grid%cam_abs_freq_s                         &
      &        ,XTIME=grid%xtime                                                &
               ,CURR_SECS=curr_secs, ADAPT_STEP_FLAG=adapt_step_flag       &
+#if ( EM_CORE == 1)
      &        ,mut=grid%mut,dnw=grid%dnw                                  & ! FARMS coupling
      &        ,swdown2=grid%swdown2, swddni2=grid%swddni2                 & ! FARMS coupling
      &        ,swddif2=grid%swddif2, swddir2=grid%swddir2                 & ! FARMS coupling
      &        ,swdownc2=grid%swdownc2, swddnic2=grid%swddnic2             & ! FARMS coupling
+#endif
```
Comment (weiwangncar) [inline dyn_em/module_first_rk_step_part1.F]: @davegill Why is this needed since this is only called from EM_CORE?
